src/main/java/core/basesyntax/db/StorageImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageImpl implements Storage {
+    private Map<String, Integer> fruitMap;
+
+    public StorageImpl() {
+        fruitMap = new HashMap<>();
+    }
+
+    @Override
+    public void add(String fruit, int quantity) {
+        fruitMap.put(fruit, quantity);
+    }
+
+    @Override
+    public void change(String fruit, int newQuantity) {
+        fruitMap.replace(fruit, newQuantity);
+    }
+
+    @Override
+    public int get(String fruit) {
+        return fruitMap.get(fruit);
+    }
+
+    @Override
+    public Map<String, Integer> getAll() {
+        return new HashMap<>(fruitMap);
```

Comment:
Returning a new HashMap instance in the `getAll` method is unnecessary and could lead to performance issues with a large map. You can return the existing map since the `Storage` class should be the only class responsible for modifying the map. If you want to prevent modification, consider using `Collections.unmodifiableMap(fruitMap)`.